front/src/modules/trainschedule/components/ManageTrainSchedule/TrainSettings.tsx

```diff
@@ -47,30 +49,20 @@ export default function TrainSettings() {
   const [name, setName] = useState<string>(nameFromStore);
   const [startTime, setStartTime] = useState(formatLocalDateTime(startTimeFromStore));
   const [initialSpeed, setInitialSpeed] = useState<number | undefined>(initialSpeedFromStore);
-  const [trainCategory, setTrainCategory] = useState<CategoryOption | null>(
-    categoryFromStore
-      ? {
-          id: categoryFromStore,
-          label: t(`categoriesOptions.${categoryFromStore}`),
-        }
-      : null
-  );
-  const dispatch = useAppDispatch();
+  const [trainCategory, setTrainCategory] = useState<CategoryOption | undefined>(undefined);
```

Comment:
You can remove the 2 `undefined` (it's implicit and makes the code easier to read)
```suggestion
  const [trainCategory, setTrainCategory] = useState<CategoryOption>();
```

Reply:
Done !